runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -194,6 +194,13 @@ public boolean workIsFailed() {
     return Optional.ofNullable(work).map(Work::isFailed).orElse(false);
   }
 
+  public byte[] getCurrentRecordOffset() {
+    if (activeReader == null) {
+      return new byte[0];
```

Comment:
have an static singleton of this?
NO_RECORD_OFFSET

or perhaps better to make this Nullable and return null? If caller expects a reader to be present this would allow for them to check that instead of possibly using the bad offset.

Reply:
I used the static singleton.

I previously tried annotating as nullable but received errors that it is not compatible with primitive types like byte. Without nullable annotation, returning null is also marked as an error. I have not done a lot of Beam Java coding, so I possibly misunderstood something.

However, I think empty bytes can/should be considered invalid, so it's okay to use empty as invalid rather than both empty and null. Wdyt?